src/ObjectUtils/PdbDiaUtils.cpp

```diff
@@ -0,0 +1,362 @@
+  }
+  if (BOOL is_unaligned; SUCCEEDED(type->get_unalignedType(&is_unaligned)) && is_unaligned) {
+    type_modifiers.push_back("__unaligned");
+  }
+  if (BOOL is_restricted; SUCCEEDED(type->get_restrictedType(&is_restricted)) && is_restricted) {
+    type_modifiers.push_back("restricted");
+  }
+
+  return absl::StrJoin(type_modifiers, " ");
+}
+
+[[nodiscard]] ErrorMessageOr<std::string> GetPointerTypeAsString(
+    IDiaSymbol* type, const std::string& parent_pointer_type_str) {
+  CComPtr<IDiaSymbol> base_type = nullptr;
+  if (FAILED(type->get_type(&base_type))) {
+    return ErrorMessage(orbit_base::GetLastErrorAsString());
+  }
+  if (base_type == nullptr) {
+    return ErrorMessage("Unable to retrieve type symbol.");
+  }
+
+  std::string new_pointer_type_str;
+  if (BOOL is_reference; SUCCEEDED(type->get_reference(&is_reference)) && is_reference) {
+    new_pointer_type_str = "&";
+  } else if (BOOL is_rvalue_reference;
+             SUCCEEDED(type->get_RValueReference(&is_rvalue_reference)) && is_rvalue_reference) {
+    new_pointer_type_str = "&&";
+  } else if (BOOL is_ptr_to_member_function;
+             SUCCEEDED(type->get_isPointerToMemberFunction(&is_ptr_to_member_function)) &&
+             is_ptr_to_member_function) {
+    CComPtr<IDiaSymbol> class_parent = nullptr;
+    if (FAILED(type->get_classParent(&class_parent))) {
+      return ErrorMessage(orbit_base::GetLastErrorAsString());
+    }
+    if (class_parent == nullptr) {
+      return ErrorMessage("Unable to retrieve class parent symbol.");
+    }
+    OUTCOME_TRY(auto&& class_parent_str, orbit_object_utils::PdbDiaTypeAsString(*&class_parent));
+    new_pointer_type_str = class_parent_str + "::*";
+  } else if (BOOL is_ptr_to_member_function;
+             SUCCEEDED(type->get_isPointerToDataMember(&is_ptr_to_member_function)) &&
+             is_ptr_to_member_function) {
+    CComPtr<IDiaSymbol> class_parent = nullptr;
+    if (FAILED(type->get_classParent(&class_parent))) {
+      return ErrorMessage(orbit_base::GetLastErrorAsString());
+    }
+    if (class_parent == nullptr) {
+      return ErrorMessage("Unable to retrieve class parent symbol.");
+    }
+    OUTCOME_TRY(auto&& class_parent_str, orbit_object_utils::PdbDiaTypeAsString(*&class_parent));
+    new_pointer_type_str = class_parent_str + "::*";
+  } else {
+    new_pointer_type_str = "*";
+  }
+
+  std::string type_modifiers = GetTypeModifiersAsString(type);
+  if (!type_modifiers.empty()) {
+    new_pointer_type_str += " ";
+    new_pointer_type_str += type_modifiers;
+  }
+
+  new_pointer_type_str += parent_pointer_type_str;
+  return orbit_object_utils::PdbDiaTypeAsString(*&base_type, new_pointer_type_str);
+}
+
+}  //  namespace
+
+namespace orbit_object_utils {
+
+ErrorMessageOr<std::string> PdbDiaParameterListAsString(IDiaSymbol* function_or_function_type) {
+  DWORD tag;
+  if (FAILED(function_or_function_type->get_symTag(&tag))) {
+    return ErrorMessage(orbit_base::GetLastErrorAsString());
```

Comment:
nit. maybe add the name of the function that failed in the ErrorMessage: 

```return ErrorMessage(absl::StrFormat("Error calling \"get_symTag\": %s", orbit_base::GetLastErrorAsString());```

same in other occurrences.

Reply:
Done.